plugin/core/sessions.py

```diff
@@ -1650,7 +1652,8 @@ def _template_variables(self) -> dict[str, str]:
         return variables
 
     def execute_command(
-        self, command: ExecuteCommandParams, progress: bool, view: sublime.View | None = None
+        self, command: ExecuteCommandParams, *, progress: bool, view: sublime.View | None = None,
+        is_refactoring: bool = False,
```

Comment:
Using `*` here is technically a breaking change but my searching shows that nobody uses it currently in a way that would cause it to break.

Feel free to double check (especially third party LSP packages).

Reply:
I think `progress` should have a default value (preferrably `False`).

I just recently used this a few times in a new plugin https://github.com/sublimelsp/LSP-Tinymist/blob/main/plugin.py and was surprised that this function requires passing at least two arguments. That code would break when changing to keyword-only arguments, but I can easily adjust it if all keyword arguments get default values.

I agree that it feels "right" that the other arguments should be keyword-only arguments.